qa/tasks/vstart_runner.py

```diff
@@ -1201,11 +1201,64 @@ def __del__(self):
         if test_path:
             shutil.rmtree(test_path)
 
+
+class LogRotate():
+    def __init__(self):
+        self.remove_old_conf_and_state_files()
+
+        self.conf_path = os.path.join(os.getcwd(), 'out/logrotate.conf')
+        self.state_file_path = os.path.join(os.getcwd(), 'out/logrotate.state')
+
+        self.write_logrotate_conf()
+        self.run_logrotate()
+
+    def remove_old_conf_and_state_files(self):
+        for file_ in ['out/logrotate.conf', 'out/logrotate.state']:
+            cmdargs = 'rm ' + os.path.join(os.getcwd(), file_)
+            remote.run(args=cmdargs, check_status=False)
+
+    def run_logrotate(self):
+        remote.run(args=['logrotate', self.conf_path, '-s',
+                         self.state_file_path, '--verbose'])
+
+    def write_logrotate_conf(self):
+        conf_contents = '''{mon_logs}
+{osd_logs}
+{mds_logs}
+{mgr_logs}
+{{
+    copytruncate
+    missingok
+    size 2G
+    rotate 20
+    prerotate
+        kill -STOP $(ps -e | grep ceph-mon | awk '{{print $1}}')
+        kill -STOP $(ps -e | grep ceph-osd | awk '{{print $1}}')
+        kill -STOP $(ps -e | grep ceph-mgr | awk '{{print $1}}')
+        kill -STOP $(ps -e | grep ceph-mds | awk '{{print $1}}')
+    endscript
+    postrotate
+        kill -CONT $(ps -e | grep ceph-mon | awk '{{print $1}}')
+        kill -CONT $(ps -e | grep ceph-osd | awk '{{print $1}}')
+        kill -CONT $(ps -e | grep ceph-mgr | awk '{{print $1}}')
+        kill -CONT $(ps -e | grep ceph-mds | awk '{{print $1}}')
+    endscript
+}}
+    '''.format(mon_logs=os.path.join(os.getcwd(), 'out/mon.*.log'),
+               osd_logs=os.path.join(os.getcwd(), 'out/osd.*.log'),
+               mgr_logs=os.path.join(os.getcwd(), 'out/mgr.*.log'),
+               mds_logs=os.path.join(os.getcwd(), 'out/mds.*.log'))
```

Comment:
I was thinking this is config could be generated by vstart.sh so anyone could run:

`logrotate ...`

in their `build/` directory to rotate the logs, not just when they're using `vstart_runner.py`. Let's do that?

Reply:
Makes sense.